src/errorline.jl

```diff
@@ -0,0 +1,236 @@
+        groupcolor = repeat(groupcolor, size(y,3)) # Use the same color for all groups
+    elseif (groupcolor !== nothing && ndims(y) > 2) && length(groupcolor) < size(y,3)
+        error("$(length(groupcolor)) colors given for a matrix with $(size(y,3)) groups")
+    end
+
+    if errorstyle == :spaghetti && numsecondarylines > size(y,2) # Override numsecondarylines
+        numsecondarylines = size(y,2)
+    end
+
+    for g = axes(y,3) # Iterate through 3rd dimension
+        # Compute center and distribution for each value of x
+        y_central, y_error = compute_error(y[:,:,g], centertype, errortype, percentiles)
+        if errorstyle == :ribbon
+            seriestype := :path
+            @series begin
+                x := x
+                y := y_central
+                ribbon := y_error
+                fillalpha --> .1
+                if groupcolor !== nothing
+                    linecolor := groupcolor[g]
+                    fillcolor := groupcolor[g]
+                end
+                () # Supress implicit return
+            end
+
+        elseif errorstyle == :stick
+            x_offset = (extrema(x)[2] - extrema(x)[1]) * stickwidth
+            seriestype := :path
+            for (i, xi) in enumerate(x)
+                # Error sticks
+                @series begin
+                    primary := false
+                    x := [xi-x_offset, xi+x_offset, xi, xi, xi+x_offset, xi-x_offset]
+                    if errortype == :percentile
+                        y := [repeat([y_central[i] - y_error[1][i]],3); repeat([y_central[i] + y_error[2][i]],3)]
+                    else
+                        y := [repeat([y_central[i] - y_error[i]],3); repeat([y_central[i] + y_error[i]],3)]
+                    end
+                    # Set the stick color
+                    if secondarycolor === nothing
+                        linecolor := :gray60
+                    elseif secondarycolor == :matched
+                        if groupcolor !== nothing
+                            linecolor := groupcolor[g]
+                        else
+                            linecolor := palette(:default)[g]
+                        end
+                    else
+                        linecolor := secondarycolor
+                    end
+                    () # Supress implicit return
+                end
+            end
+
+            # Base line
+            seriestype := :line
+            @series begin
+                primary := true
+                x := x
+                y := y_central
+                if groupcolor !== nothing
+                    linecolor := groupcolor[g]
+                end
+                ()
+            end
+
+        elseif errorstyle == :spaghetti
+            num_obs = size(y,2)
+            if num_obs > numsecondarylines
+                sub_sample_idx = sample(1:num_obs, numsecondarylines, replace=false)
+                y_sub_sample = y[:,sub_sample_idx,g]
```

Comment:
I don't know what convention is used in `StatsPlots` (maybe we need an autoformatter here as well), but I find `y[:, sub_sample_idx, g]` more readable.